src/SymbolCollector.Core/Client.cs

```diff
@@ -107,62 +114,138 @@ private async Task UploadAsync(string debugId, string file, CancellationToken ca
                     }
+                    continue;
+                }
 
-                    var hasUnwindingInfo = elf.TryGetSection(".eh_frame", out _);
-                    var hasDwarfDebugInfo = elf.TryGetSection(".debug_frame", out _);
+                yield return (buildId, file);
+            }
+        }
 
-                    if (!hasUnwindingInfo && !hasDwarfDebugInfo)
+        private string? GetMachOBuildId(string file)
+        {
+            try
+            {
+                // TODO: find an async API if this is used by the server
+                if (MachOReader.TryLoad(file, out _) == MachOResult.OK)
+                {
+                    using var algorithm = SHA256.Create();
+                    var hash = algorithm.ComputeHash(File.ReadAllBytes(file));
+                    var builder = new StringBuilder();
+                    foreach (var b in hash)
                     {
-                        _logger.LogWarning("No unwind nor DWARF debug info in {file}", file);
-                        continue;
+                        builder.Append(b.ToString("x2"));
                     }
 
-                    _logger.LogInformation("Contains unwinding info: {hasUnwindingInfo}", hasUnwindingInfo);
-                    _logger.LogInformation("Contains DWARF debug info: {hasDwarfDebugInfo}", hasDwarfDebugInfo);
+                    return builder.ToString();
+                }
 
-                    var builder = new StringBuilder();
-                    var bytes = buildId.GetContents().Skip(16);
+                _logger.LogWarning("Couldn't load': {file} with mach-O reader.", file);
+            }
+            catch (Exception e)
+            {
+                // You would expect TryLoad doesn't throw but that's not the case
+                _logger.LogError(e, "Failed processing file {file}.", file);
+            }
 
-                    foreach (var @byte in bytes)
-                    {
-                        builder.Append(@byte.ToString("x2"));
-                    }
+            return null;
+        }
 
-                    buildIdHex = builder.ToString();
-                }
-                catch (Exception e)
+        private string? GetElfBuildId(string file)
+        {
+            IELF? elf = null;
+            try
+            {
+                // TODO: find an async API if this is used by the server
+                if (ELFReader.TryLoad(file, out elf))
                 {
-                    // You would expect TryLoad doesn't throw but that's not the case
-                    _logger.LogError(e, "Failed processing file {file}.", file);
-                    continue;
+                    var hasBuildId = elf.TryGetSection(".note.gnu.build-id", out var buildId);
+                    if (hasBuildId)
+                    {
+                        var hasUnwindingInfo = elf.TryGetSection(".eh_frame", out _);
+                        var hasDwarfDebugInfo = elf.TryGetSection(".debug_frame", out _);
```

Comment:
This does not necessarily indicate that there is full debug information. .debug_frame is essentially the same as .eh_frame.

Reply:
Right now I'm just logging this out. Whether the object had those sections or not. We upload them regardless, any ELF or Mach-O file is uploaded.

> This does not necessarily indicate that there is full debug information.
What would indicate full debug info?

> .debug_frame is essentially the same as .eh_frame.
Why are they essentially the same? I notice some objects have one but not the other?